suite-common/wallet-core/src/device/deviceReducer.ts

```diff
@@ -700,7 +700,19 @@ export const prepareDeviceReducer = createReducerWithExtraDeps(initialState, (bu
 // Basic state selectors (no need to wrap in createMemoizedSelector)
 export const selectDevices = (state: DeviceRootState) => state.device?.devices;
 export const selectDevicesCount = (state: DeviceRootState) => state.device?.devices?.length;
-export const selectSelectedDevice = (state: DeviceRootState) => state.device.selectedDevice;
+export const selectSelectedDevice = (state: DeviceRootState) => {
+    const deviceId = state.device?.selectedDevice?.id;
+    const devicePath = state.device?.selectedDevice?.path;
+    const deviceInstance =
+        state.device?.selectedDevice?.instance ?? state.device?.selectedDevice?.walletNumber;
+
+    return state.device?.devices.find(
+        device =>
+            device.id === deviceId &&
+            device.path === devicePath &&
+            (device.instance === deviceInstance || device.walletNumber === deviceInstance),
+    );
+};
```

Comment:
@vojtatranta why this change? I agree that device.selectedDevice should only be a pointer to device.devices. But it should be refactored everywhere and in a single blow. This indicated that there is a problem with updating selected device and you are trying to solve it this way?

Reply:
I put it back

Reply:
key, checking